static/app/views/dashboardsV2/widgetCard/widgetQueries.tsx

```diff
@@ -115,31 +115,37 @@ function WidgetQueries({
   const config = ErrorsAndTransactionsConfig;
   const context = useContext(DashboardsMEPContext);
 
-  let isMetricsData: undefined | boolean;
   let setIsMetricsData: undefined | ((value?: boolean) => void);
 
   if (context) {
-    isMetricsData = context.isMetricsData;
     setIsMetricsData = context.setIsMetricsData;
   }
 
-  const afterFetchSeriesData = useCallback(
-    (rawResults: SeriesResult) => {
-      // If one of the queries is sampled, then mark the whole thing as sampled
-      const currentResultIsMetricsData =
-        isMetricsData === false ? false : getIsMetricsDataFromSeriesResponse(rawResults);
-      setIsMetricsData?.(currentResultIsMetricsData);
-    },
-    [isMetricsData, setIsMetricsData]
-  );
+  const isSeriesMetricsDataResults: boolean[] = [];
+  const afterFetchSeriesData = (rawResults: SeriesResult) => {
+    if (rawResults.data) {
+      if (rawResults.isMetricsData !== undefined) {
+        isSeriesMetricsDataResults.push((rawResults as EventsStats).isMetricsData!);
+      }
+    } else {
```

Comment:
```suggestion
    if (rawResults.data) {
      rawResults = rawResults as EventsStats;
      if (rawResults.isMetricsData !== undefined) {
        isSeriesMetricsDataResults.push(rawResults.isMetricsData);
      }
    } else {
```

Just an idea, I was trying to get rid of the `!` and make it more explicit that after the first check for `rawResults.data`, we know it's an `EventsStats` type. I just think `!` isn't something we want to have a lot of if we can avoid it

Reply:
Yeah this is much cleaner 😄